src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ScheduleRuleService.kt

```diff
@@ -42,26 +40,29 @@ class ScheduleRuleService {
 
   fun buildNewRules(requestedRules: List<ScheduleRuleRequest>, existingActivity: CourseActivity): List<CourseScheduleRule> {
     val monthStart = LocalDate.now().with(TemporalAdjusters.firstDayOfMonth())
-    return requestedRules.map {
-      validateRequest(it)
-      CourseScheduleRule(
-        id = findExistingRuleId(it, existingActivity.courseScheduleRules),
-        courseActivity = existingActivity,
-        startTime = LocalDateTime.of(monthStart, it.startTime),
-        endTime = LocalDateTime.of(monthStart, it.endTime),
-        slotCategory = SlotCategory.of(it.startTime),
-        monday = it.monday,
-        tuesday = it.tuesday,
-        wednesday = it.wednesday,
-        thursday = it.thursday,
-        friday = it.friday,
-        saturday = it.saturday,
-        sunday = it.sunday,
-      )
+    return requestedRules.map { request ->
+      validateRequest(request)
+      findExistingRule(request, existingActivity.courseScheduleRules)
```

Comment:
Need to use the same object if it already exists, otherwise Hibernate gets confused